src/main/java/redis/clients/jedis/Connection.java

```diff
@@ -88,24 +89,34 @@ protected Connection sendCommand(final ProtocolCommand cmd, final String... args
     return sendCommand(cmd, bargs);
   }
 
-  protected Connection sendCommand(final ProtocolCommand cmd, final byte[]... args) {
-    try {
-      connect();
-      Protocol.sendCommand(outputStream, cmd, args);
-      return this;
-    } catch (JedisConnectionException ex) {
-      // Any other exceptions related to connection?
-      broken = true;
-      throw ex;
-    }
+  protected Connection sendCommand(final ProtocolCommand cmd) {
+    return sendCommand(cmd, EMPTY_ARGS);
   }
 
-  protected Connection sendCommand(final ProtocolCommand cmd) {
+  protected Connection sendCommand(final ProtocolCommand cmd, final byte[]... args) {
     try {
       connect();
-      Protocol.sendCommand(outputStream, cmd, new byte[0][]);
+      Protocol.sendCommand(outputStream, cmd, args);
       return this;
     } catch (JedisConnectionException ex) {
+      /*
+       * When client send request which formed by invalid protocol, Redis send back error message
+       * before close connection. We try to read it to provide reason of failure.
+       */
+      try {
+        // Set soTimeout 100ms to avoid long blocking
+        socket.setSoTimeout(100);
+        String errorMessage = Protocol.readErrorLineIfPossible(inputStream);
```

Comment:
I didn't use `Protocol.read(inputStream)` because it throw expection.
At this context read error message from InputStream is normal case, so your approach is work but not properly expose intention.
Though there is duplication between `processError` and `readErrorLineIfPossible` but I thought it's worth.
What do you think?